repo/content/index_blob_manager.go

```diff
@@ -311,17 +300,7 @@ func (m *indexBlobManagerImpl) deleteOldBlobs(ctx context.Context, latestBlob bl
 		return errors.Wrap(err, "unable to schedule delayed cleanup of blobs")
 	}
 
-	if err := m.deleteBlobsFromStorageAndCache(ctx, compactionLogBlobsToDelete); err != nil {
-		return errors.Wrap(err, "unable to delete compaction logs")
-	}
-
-	if err := m.deleteBlobsFromStorageAndCache(ctx, cleanupBlobsToDelete); err != nil {
-		return errors.Wrap(err, "unable to delete cleanup blobs")
-	}
-
-	m.flushCache()
-
-	return nil
+	return m.cleanupCleanupBlobs(ctx, latestBlob.Timestamp)
```

Comment:
It seems that there is a subtle interaction here that could prevent both cleanup and compaction logs from being deleted.
- `delayCleanupBlobs` above may "re-create" or change the time stamp of already existing cleanup blobs.
- then, when `cleanupCleanupBlobs` executes, it will list the cleanup blobs and all the ones that were (re) created above will have a recent time stamp and will not get deleted, even though they may be old enough. It seems that the same would happen in following iterations, so the cleanup blobs would be touched again and compaction blobs would remain.
Or maybe I'm missing something.

Reply:
i think you're correct, let me think about it more

Reply:
refactored log entries to:

* have all blobIDs listed together
* include timestamp of last write

this ensures that cleanup blobs will never be overwritten.